packages/perseus-editor/src/widgets/__tests__/free-response-editor.test.tsx

```diff
@@ -31,6 +31,56 @@ describe("free-response editor", () => {
         expect(screen.getByDisplayValue("test-question")).toBeInTheDocument();
     });
 
+    it("calls onChange when the allow unlimited characters is changed", async () => {
+        // Arrange
+        const onChangeMock = jest.fn();
+
+        // Act
+        render(
+            <FreeResponseEditor
+                allowUnlimitedCharacters={false}
+                onChange={onChangeMock}
+            />,
+        );
+
+        await userEvent.click(
+            screen.getByRole("checkbox", {name: /Allow unlimited characters/i}),
+        );
+
+        // Assert
+        expect(onChangeMock).toBeCalledWith({allowUnlimitedCharacters: true});
+    });
+
+    it("calls onChange when the character limit is changed to a number", async () => {
+        // Arrange
+        const onChangeMock = jest.fn();
+
+        // Act
+        render(
+            <FreeResponseEditor characterLimit={0} onChange={onChangeMock} />,
+        );
+
+        await userEvent.type(screen.getByLabelText(/Character limit/), "1");
+
+        // Assert
+        expect(onChangeMock).toBeCalledWith({characterLimit: 1});
+    });
+
+    it("does not call onChange when the character limit is changed to a non-number", async () => {
+        // Arrange
+        const onChangeMock = jest.fn();
+
+        // Act
+        render(
+            <FreeResponseEditor characterLimit={5} onChange={onChangeMock} />,
+        );
+
+        await userEvent.type(screen.getByLabelText(/Character limit/), "e");
```

Comment:
I mean `e` [is a number](https://en.wikipedia.org/wiki/E_(mathematical_constant)) 😆 

(ignore me)

Reply:
2.718 character limit 😂